startup.py

```diff
@@ -0,0 +1,340 @@
+
+        return  sw_versions
+
+
+    def prepare_launch(self, exec_path, args, options, file_to_open=None):
+        """
+        Prepares the given software for launch
+
+        :param exec_path: Path to DCC executable to launch
+        :param args: Command line arguments as strings
+        :param options: DCC specific options to pass
+        :param file_to_open: (Optional) Full path name of a file to open on launch
+
+        :returns: LaunchInformation instance
+        """
+        required_env = {}
+        startup_path = os.path.join(self.disk_location, "startup")
+        sgtk.util.append_path_to_env_var("PYTHONPATH", startup_path)
+        required_env["PYTHONPATH"] = os.environ["PYTHONPATH"]
+        required_env["TANK_ENGINE"] = self.engine_name
+        required_env["TANK_CONTEXT"] = sgtk.context.serialize(self.context)
+        if file_to_open:
+            required_env["TANK_FILE_TO_OPEN"] = file_to_open
+
+        return LaunchInformation(exec_path, args, required_env)
+
+
+def _icon_from_executable(exec_path):
+    """
+    Find the application icon based on the executable path and
+    current platform.
+
+    :param exec_path: Full path to the executable
+
+    :returns: Full path to application icon as a string or None.
+    """
+    icon_base_path = ""
+    if sys.platform == "darwin" and "Maya.app" in exec_path:
+        # e.g. /Applications/Autodesk/maya2016.5/Maya.app/Contents
+        icon_base_path = os.path.join(
+            "".join(exec_path.partition("Maya.app")[0:2]),
+            "Contents"
+        )
+
+    elif sys.platform in ["win32", "linux2"] and "bin" in exec_path:
+        # e.g. C:\Program Files\Autodesk\Maya2017\  or
+        #      /usr/autodesk/maya2017/
+        icon_base_path = "".join(exec_path.partition("bin")[0:1])
+
+    if not icon_base_path:
+        # If no base path, no icon
+        return None
+
+    # Append the standard icon to the base path and
+    # return that path if it exists, else None.
+    icon_path = os.path.join(icon_base_path, "icons", "mayaico.png")
+    return icon_path if os.path.exists(icon_path) else None
+
+
+def _synergy_config_files():
+    """
+    Scans the local file system using a list of search paths for
+    Autodesk Synergy Config files (.syncfg).
+
+    :returns: List of path names to Synergy Config files found
+              in the local environment
+    """
+    # Check for custom paths defined by the SYNHUB_CONFIG_PATH env var.
+    env_paths = os.environ.get("SYNHUB_CONFIG_PATH")
+    search_paths = []
+    if isinstance(env_paths, basestring):
+        # This can be a list of directories and/or files.
+        search_paths = env_paths.split(os.pathsep)
+
+    # Check the platfom-specific default installation path
+    # if no paths were set in the environment
+    elif sys.platform == "darwin":
+        search_paths = ["/Applications/Autodesk/Synergy/"]
+    elif sys.platform == "win32":
+        search_paths = ["C:\\ProgramData\\Autodesk\\Synergy\\"]
```

Comment:
`[r"C:\ProgramData\Autodesk\Synergy"]`

You won't be able to put a backslash on the end, even using a raw string, but I don't think you need to anyway -- same above with the darwin path -- since os.path.join below will take care of it.